lib/tests/test_shape_utils.py

```diff
@@ -140,3 +146,47 @@ def test_drop_parts_of_geoms(self, capsys, europe_gdf):
             europe_gdf[europe_gdf.name == 'Highlands and Islands'].area
             > gdf[gdf.name == 'Highlands and Islands'].area
         ).all()
+
+class TestPolygonsFromPoints:
+
+    @pytest.fixture
+    def points(self):
+        points = {  # Latitiude, Longitude
+            'athens': [37.9838, 23.7275],
+            'davos': [46.8027, 9.8360],
+            'stockholm': [59.3293, 18.0686],
+            'wurzburg': [49.7913, 9.9534]
+        }
+        areas = {
+            'athens': 1,
+            'davos': 5,
+            'stockholm': 100,
+            'wurzburg': 500
+        }
+        point_gdf = gpd.GeoDataFrame(
+            # x = longitude, y = latitude
+            geometry=gpd.points_from_xy([i[1] for i in points.values()], [i[0] for i in points.values()]),
+            index=points.keys(),
+            crs = WGS84
+        )
+        point_gdf["REP_AREA"] = pd.Series(areas)
+        return point_gdf
+
+    @pytest.fixture
+    def estimated_polygons(self, points):
+        return estimate_polygons_from_points(points, "REP_AREA")
```

Comment:
Hmm, I would say this is part of the arranging process. As given in the example of what fixtures are, the calls to the class being tested are all part of fixtures. This is the same process I'm undertaking here. I then only assert in the test.

Reply:
I don't see it that way. `estimate_polygons_from_points ` is the behaviour/functionality you are testing, so it belongs into `act` rather than `arrange`.

Reply:
hmm, ok. I can remove it, but it makes sense as a fixture as it has three tests associated with the same function call. Better to do that once than thrice...

Reply:
Ok, I checked the pytest doc a little longer and it seems we are both partly right according to their definitions:

> In pytest, “fixtures” are functions you define that serve this purpose. But they don’t have to be limited to just the arrange steps. They can provide the act step, as well, and this can be a powerful technique for designing more complex tests, especially given how pytest’s fixture system works. But we’ll get into that further down.

See this example: https://docs.pytest.org/en/stable/fixture.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached